packages/fastify-vite/config.js

```diff
@@ -232,31 +234,41 @@ function resolveClientModule(root) {
   return null
 }
 
-function resolveRoot(path) {
+function resolveRoot(distDir, path) {
   let root = path
   if (root.startsWith('file:')) {
     root = fileURLToPath(root)
   }
+  let dir = root
   if (stat(root).isFile()) {
-    return dirname(root)
+    dir = dirname(root)
   }
-  return root
+  // Check if running under dist folder
+  const split = dir.split(sep)
+  if (split.at(-1) === distDir) {
```

Comment:
I'm a little confused here. Isn't it possible for `distDir` to be an absolute path with lots of slashes?

Reply:
Good catch - internally the `distDir` `@fastify/vite` option needs to be resolved as well.